pkg/utils/operation.go

```diff
@@ -41,7 +41,7 @@ import (
 // WorkflowOperator is operation handler for workflow's suspend/resume/rollback/restart/terminate
 type WorkflowOperator interface {
 	Suspend(ctx context.Context) error
-	Resume(ctx context.Context) error
+	Resume(ctx context.Context, step string) error
```

Comment:
It's not good practice to change an interface, we can probably add a new:

```
ResumeStep(ctx context.Context, stepName string) error
```

Reply:
The caller will actually wrap these two functions in one for users to use(whether it is to continue the workflow or to specify the steps), so I think it is better to combine the two functions.

Reply:
Actually you're breaking the princples of interface here:

1. **Interface should be simple, intuitive**.  It's hard to learn that specify an empty step means resume the whole workflow.
2. **Interface should be consisten**. You're promising the functions will be consitent and compatible when you declare an interface. While you're breaking it now. Assume you already have other users implementing this interface in different ways, you're forcing them to align the changes with you, that could be huge of work.

What the best practice? 

We can decalre an other interface, maybe called `WorkflowStepOperator` that can do work for specified step.  The implementation below can both adopt the `WorkflowStepOperator` and `WorkflowOperator` interfaces while the consistency of the `WorkflowOperator` interface can be promised.